### ISSUE TITLE
If blocks that only declare variables are omitted

### ISSUE THREAD

**Issue by leedm777** [rollup/rollup#997]
Consider the following code:

``` javascript
// bad.js
if (process.env.NODE_ENV !== 'production') {
  var foo = function() {
    console.log('Yeah, whatever');
  };
}

function doSomething() {
  foo();
}

if (process.env.NODE_ENV !== 'production') {
  doSomething();
}
```

When rollup generates code, the `if` block gets omitted:

``` javascript
// bad.bundle.js
function doSomething() {
  foo();
}

if (process.env.NODE_ENV !== 'production') {
  doSomething();
}
```

So while this code might not follow best practices, because of [var hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting) is is _technically_ correct. This sort of thing [does happen](https://github.com/facebook/react/blob/178cb7d3392505cd393b68e4e061c6ef860344a1/src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js), and can cause problems if one pulls in a library that doesn't follow best practices.

**Comment by brianmhunt** [comment]
Aye, live example: We're getting this with knockout/tko (github.com/knockout/tko.binding.core/src/textInput).

I think Rollup should respect var hoisting and leave the if block in.  Sure, it's bad practice, but it's also hard to root out every instance in a large legacy codebase (like Knockout's), and can leave some nasty surprises for folks that don't expect it.

In contrast, blocks with just `let` and `const` should be omitted.

**Comment by Rich-Harris** [comment]
Fixed in 0.38.2. thanks
